### ISSUE TITLE
Migrations failing on latest Docker image

### ISSUE THREAD

**Issue by couquino** [n8n-io/n8n#3826]
**Describe the bug**
Just after the latest update of the Docker image (n8nio/n8n:latest@sha256:3b9eb2c6996bbae3044033a4bc4c18108b6de218562ded2ae3f36f9d9aa66980), the container is unhealthy and I now have these errors in logs

```
2022-08-03T19:05:25+02:00	/usr/local/lib/node_modules/n8n/dist/src/databases/migrations/mysqldb/1658932910559-AddNodeIds.js:43
2022-08-03T19:05:17+02:00	    at async AddNodeIds1658932910559.up (/usr/local/lib/node_modules/n8n/dist/src/databases/migrations/mysqldb/1658932910559-AddNodeIds.js:40:9)
2022-08-03T19:05:17+02:00	    at processTicksAndRejections (node:internal/process/task_queues:96:5)
2022-08-03T19:05:17+02:00	    at runChunked (/usr/local/lib/node_modules/n8n/dist/src/databases/utils/migrationHelpers.js:75:15)
2022-08-03T19:05:17+02:00	    at /usr/local/lib/node_modules/n8n/dist/src/databases/migrations/mysqldb/1658932910559-AddNodeIds.js:41:23
2022-08-03T19:05:17+02:00	    at Array.forEach (<anonymous>)
2022-08-03T19:05:17+02:00	    at /usr/local/lib/node_modules/n8n/dist/src/databases/migrations/mysqldb/1658932910559-AddNodeIds.js:43:23
2022-08-03T19:05:17+02:00	TypeError: nodes.forEach is not a function
2022-08-03T19:05:17+02:00	                      ^
2022-08-03T19:05:17+02:00	                nodes.forEach((node) => {
2022-08-03T19:05:17+02:00	/usr/local/lib/node_modules/n8n/dist/src/databases/migrations/mysqldb/1658932910559-AddNodeIds.js:43
```

**To Reproduce**
1. Update the Docker image of N8N to n8nio/n8n:latest@sha256:3b9eb2c6996bbae3044033a4bc4c18108b6de218562ded2ae3f36f9d9aa66980

**Expected behavior**
The migration should execute fine

**Environment (please complete the following information):**
 - OS: Docker (Ubuntu 22.04 LTS host)
 - n8n Version: n8nio/n8n:latest@sha256:3b9eb2c6996bbae3044033a4bc4c18108b6de218562ded2ae3f36f9d9aa66980
 - Node.js Version: 16.16.0
 - Database system: mysqldb
 - Operation mode: N/A

**Additional context**

**Comment by mutdmour** [comment]
Thanks for sharing. You seem to have an invalid workflow obj in your db. Would you be able to look into the db and tell us if the nodes object is `undefined` or not an array? In the meantime, will prepare a fix for this.

**Comment by couquino** [comment]
I just checked my DB and the `workflow_entity` table seems to be correct (I validated the json of "nodes" column)
However, the `installed_nodes` table is empty.

The migration table contains these migrations:

'1', '1588157391238', 'InitialMigration1588157391238'
'2', '1592447867632', 'WebhookModel1592447867632'
'3', '1594902918301', 'CreateIndexStoppedAt1594902918301'
'4', '1607431743767', 'MakeStoppedAtNullable1607431743767'
'5', '1611149998770', 'AddWebhookId1611149998770'
'6', '1615306975123', 'ChangeDataSize1615306975123'
'7', '1617268711084', 'CreateTagEntity1617268711084'
'8', '1620729500000', 'ChangeCredentialDataSize1620729500000'
'9', '1620826335440', 'UniqueWorkflowNames1620826335440'
'10', '1623936588000', 'CertifyCorrectCollation1623936588000'
'11', '1626183952959', 'AddWaitColumnId1626183952959'
'12', '1630451444017', 'UpdateWorkflowCredentials1630451444017'
'13', '1644424784709', 'AddExecutionEntityIndexes1644424784709'
'14', '1646992772331', 'CreateUserManagement1646992772331'
'15', '1648740597343', 'LowerCaseUserEmail1648740597343'
'16', '1652367743993', 'AddUserSettings1652367743993'
'17', '1652905585850', 'AddAPIKeyColumn1652905585850'
'18', '1652254514003', 'CommunityNodes1652254514003'
'19', '1654090101303', 'IntroducePinData1654090101303'

**Comment by alexandervnuchkov** [comment]
I also have the same error with the latest Docker image 0.189.0 and MariaDB database. Downgrading to 0.188.0 solved the issue.
My system: host Raspbian 11 bullseye (arm64), Raspberry Pi 4 with 4 GB memory.

**Comment by mutdmour** [comment]
@couquino what version of MySQL are you running? Cannot reproduce with version 8. @alexandervnuchkov What version of MariaDB are you running?

**Comment by alexandervnuchkov** [comment]
@mutdmour 

```shell
$ mariadb --version
mariadb  Ver 15.1 Distrib 10.9.1-MariaDB, for debian-linux-gnu (aarch64) using  EditLine wrapper
```

**Comment by couquino** [comment]
@mutdmour (lscr.io/linuxserver/mariadb)
![image](https://user-images.githubusercontent.com/30592770/182830127-7642eab0-ded3-48e7-83d5-82abed3aa5fd.png)

**Comment by grahamkennery** [comment]
Also receiving the same error over here with the latest update and MariaDB:
`mariadb  Ver 15.1 Distrib 10.5.16-MariaDB, for Linux (x86_64) using readline 5.1` 

My `workflow_entity` table has a single row, the `nodes` field is a valid JSON array.

**Comment by mutdmour** [comment]
thanks was able to reproduce it and address the issue

**Comment by floryn90** [comment]
hi guys,
I'm facing the same issue:

` kubectl logs -n n8n n8n-9bd96d9bc-vz8xm --follow
Initializing n8n process
/usr/local/lib/node_modules/n8n/dist/src/databases/migrations/mysqldb/1658932910559-AddNodeIds.js:43
                nodes.forEach((node) => {
                      ^

TypeError: nodes.forEach is not a function
    at /usr/local/lib/node_modules/n8n/dist/src/databases/migrations/mysqldb/1658932910559-AddNodeIds.js:43:23
    at Array.forEach (<anonymous>)
    at /usr/local/lib/node_modules/n8n/dist/src/databases/migrations/mysqldb/1658932910559-AddNodeIds.js:41:23
    at runChunked (/usr/local/lib/node_modules/n8n/dist/src/databases/utils/migrationHelpers.js:75:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async AddNodeIds1658932910559.up (/usr/local/lib/node_modules/n8n/dist/src/databases/migrations/mysqldb/1658932910559-AddNodeIds.js:40:9)
`

Database version: "mariadb:10.8"

**Comment by janober** [comment]
Got fixed with PR #3832 and released with `n8n@0.189.``
